frontend/catalyst/jax_primitives.py

```diff
@@ -1455,6 +1457,43 @@ def _adjoint_lowering(
     return op.results
 
 
+#
+# tensor_init
+#
+
+
+@tensor_init_p.def_impl
+def _tensor_init_def_impl(
+    ctx, dynamic_shape, *, static_shape, initializer, dtype
+):  # pragma: no cover
+    raise NotImplementedError()
+
+
+@tensor_init_p.def_abstract_eval
+def _tensor_init_abstract(dynamic_shape, *, static_shape, initializer, dtype):
+    static_shape = [dim if dim >= 0 else ir.ShapedType.get_dynamic_size() for dim in static_shape]
+    return core.ShapedArray(static_shape, dtype=dtype)
+
+
+def _tensor_init_lowering(
+    jax_ctx: mlir.LoweringRuleContext, dynamic_shape: ir.Value, *, static_shape, initializer, dtype
+) -> ir.Value:
+    empty = mlir.ir_constants(True) if initializer is None else None  # do not initialize
+    output_type = [mlir.aval_to_ir_types(a)[0] for a in jax_ctx.avals_out][0]
+
+    init_value = initializer if not empty else 0
+    initializer_attr = None
+    if finfo_or_none(dtype) is not None:
+        initializer_attr = ir.FloatAttr.get(output_type.element_type, float(init_value))
+    elif iinfo_or_none(dtype) is not None:
+        initializer_attr = ir.IntegerAttr.get(output_type.element_type, int(init_value))
+    else:
+        raise ValueError(f"Unsupported initializer {initializer} of type {dtype}")
```

Comment:
Would a ComplexType also make sense here?

Reply:
Hmm. I thought we don't have support of complex arrays, do we? I need to check..